students/ThanhNhan/lesson09/mail5.py

```diff
@@ -0,0 +1,197 @@
+    #     """
+
+        #writes each letter to disk as txt file
+        #open ("./thankyou.txt", "w")
+        # tempfile.gettempdir() 
+
+        for key, values in self.data_collection.items():  
+            d = Donor(key)
+            letter = d.generate_letter(values)
+            filename = key + ".txt"
+            self.ck_dir()    
+            try:
+                print(filename) 
+                filename = os.path.join(self.DIR_NAME, filename)
+                open(filename, 'w').write(letter)
+            except FileNotFoundError:
+                print("couldn't open missing.txt")
+            finally:
+                pass
+    
+
+    def add_donor(self, Donor):
+        if self.find_donor() is True:
+            self.data_collection[Donor.name].append(Donor.donations[0])    
+        else:
+            self.data_collection[Donor.name] = Donor.donations
+
+
+
+class Donor(DonorCollection):
+
+    def __init__(self, name, donations=None):  
+        """
+        Create Donor object
+        param name: the full name of the donor
+        param donations: the donations 
+        """
+
+        self.name = name
+        if donations is None:
+            self.donations = []
+        else:
+            self.donations = [donations]
+        print("name is {}".format(self.name))
+        print("donations is", self.donations)
+
+    # @property #outside look like attribute
+    def add_donations(self, amount):
+        """
+        Add donation amount of user input
+        """
+        donate_amount = self._ck_dollar(amount)
+        self.donations.append(donate_amount)
+    
+    @property #outside look like attribute
+    def num_donations(self):
+        """
+        measure 
+        param donations: iterable donations
+        """
+        return len(self.donations)
+
+
+    @staticmethod
+    def _ck_dollar(d_amount):
+        """
+        This method is check if user input correct float number.
+        Check if user not a nubmber or infinity number
+        Check if user enter negative amount
+
+        params amount:  input amount of dollar value
+        returns amount dollar value
+        """
+
+        amount_flt = round(float(d_amount), 2)
+
+        if math.isnan(amount_flt) or math.isinf(amount_flt):
+            raise ValueError("Is not a number or infinite number")
+
+        if amount_flt <= 0.00:
+            raise ValueError("Can't be negative number")
+            
+        return amount_flt
+
+    
+    def validate_name(self):
+        """
+        This method is validate user name input
+
+        params:  input user name
+        returns true or false if name is correct input
+        """
+
+        if (self.name == " " or self.name.isdigit()):
+            return False
+
+        str_name = self.name.rstrip()
+        # #parse first and last name into string
+        try:
+            str_name = str_name.split(' ') 
+            if (str_name[1] == ' '):    
```

Comment:
I don't think split() will ever return an space for an item. 

YOu want to check how long str_name is -- that will tell you how many parts are in the name.